elfi/distributions.py

```diff
@@ -7,32 +7,34 @@
 from . import core
 
 
-def spr_op(distribution, size, input_dict):
+# TODO: combine with `simulator_wrapper` and perhaps with `summary_wrapper`?
+def random_wrapper(operation, input_dict):
     prng = npr.RandomState(0)
     prng.set_state(input_dict['random_state'])
-    size = (input_dict['n'],)+tuple(size)
-    data = distribution.rvs(*input_dict['data'], size=size, random_state=prng)
+    batch_size = input_dict["n"]
+    data = operation(*input_dict['data'], batch_size=batch_size, random_state=prng)
     return core.to_output_dict(input_dict, data=data, random_state=prng.get_state())
 
 
-class ScipyRV(core.RandomStateMixin, core.Operation):
+class Distribution:
+    """Must have an attribute or property `name`
```

Comment:
Docstring should explain the purpose of the class as well (eg. do we expect that all distributions inherit this interface?) Also as this is an interface class all the methods should have proper docstrings as well to explain expected behavior.